src/ReplaySession.cc

```diff
@@ -1018,6 +1033,8 @@ Completion ReplaySession::try_one_trace_step(
     // cont_syscall_boundary handles the ticks constraint for those cases.
     return advance_to_ticks_target(t, constraints);
   }
+  if (current_step.action != TSTEP_ENTER_SYSCALL)
+    clear_syscall_bp();
```

Comment:
I'm not sure (I though so too). This may have just hidden a different bug that I fixed completely later. I'll remove it and run the test suite to see if it works.